### PR TITLE
Script to compare if packages are updated on CI

### PR DESCRIPTION
**Pull request by andys8**
Script checks each package if there is a newer version on github.
It'll update an issue and open/close its state depending on the result.

# Related

## Original Repository

<https://github.com/andys8/purescript-update-package-sets>

## Issues

<https://github.com/purescript/package-sets/issues/704#issuecomment-707702373>

### CODE REVIEW

File: .github/workflows/check-versions.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Compare package versions
+
+on:
+  schedule:
+    - cron: "0 3 * * *"
```
Comment (andys8) [inline .github/workflows/check-versions.yaml:5]: One can add `push:` here for initial testing. In the long run, executing this script too often will likely lead to github api rate limittiing.

File: .github/workflows/check-versions.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        with:
+          key: ${{ runner.os }}-spago-${{hashFiles('ci/*.dhall')}}-${{hashFiles('ci/package-lock.json')}}
+          path: |
+            ci/node_modules
+            ci/.spago
+            ci/output
+      - run: npm --prefix ci install
+      - run: npm --prefix ci run build
+      - name: Compare package versions
+        run: npm --prefix ci start ${{ secrets.GITHUB_TOKEN }}
```
Comment (andys8) [inline .github/workflows/check-versions.yaml:23]: In my project, I created a personal token. I'm not what's the best approach here. The token needs to have the rights to update the configured (hardcoded) issue.

See https://github.com/purescript/package-sets/pull/733/files#diff-2928a06dfa0974722e5e2dba6d89c923e1bd975239b5479a7a6072f9c5f9dc68R65-R67

File: .github/workflows/check-versions.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+      - uses: actions/cache@v2
+        with:
+          key: ${{ runner.os }}-spago-${{hashFiles('ci/*.dhall')}}-${{hashFiles('ci/package-lock.json')}}
+          path: |
+            ci/node_modules
+            ci/.spago
+            ci/output
+      - run: npm --prefix ci install
```
[the remark this answers is not in the document]
Comment (andys8) [reply .github/workflows/check-versions.yaml]: Yep, changed it.

File: .github/workflows/check-versions.yaml (hunk 2 above)
Comment (andys8) [reply .github/workflows/check-versions.yaml:23]: It had not the rights to change a comment my user (@andys8) created in this repository (https://github.com/purescript/package-sets), when running as an action in (https://github.com/andys8/purescript-update-package-sets). There might be a change that it has enough when running in the same repository where the comment is. I would give it a try. If it fails, I'd create a personal token, and change the github action to use that one. One could also create a new issue, but doesn't need to.

**Comment by andys8** [comment]
Force pushed the changes an rebased onto `master`

**Comment by andys8** [comment]
I recently noticed github will stop scheduled jobs for Repositories that have been inactive for 60 days. This wouldn't be the case for this PureScript/package-sets, but for https://github.com/andys8/purescript-update-package-sets. So it would be an improvement to integrate it here with this PR.

Can I do anything to "push this over the line"?

**Comment by f-f** [comment]
@andys8 I'm so sorry, I thought we merged this already :sweat_smile: :see_no_evil: 

I'll take another quick look and merge this. Thank you again! (for the ping too)

**Review by f-f — APPROVED** [review]

**Comment by andys8** [comment]
Thanks for the review. It seems to work nicely.

Screenshot_20201219-104826

**Comment by andys8** [comment]
I deprecated the original repository, disabled CI there, and archived it.
https://github.com/andys8/purescript-update-package-sets

**Comment by f-f** [comment]
Wonderful! Thank you again 😊

**Pull request merged**